pkg/metric/collector.go

```diff
@@ -98,6 +99,11 @@ func (csi CSICollector) Collect(ch chan<- prometheus.Metric) {
 }
 
 func execute(name string, c Collector, ch chan<- prometheus.Metric) {
+	defer func() {
```

Comment:
Can we just stop collector when summary collect panic, Retry is pointless.

Reply:
The summary collector currently will not panic anymore after fix. 

Abount this snippet, just in case any individual collector panic occurred, the `execute` should  recover from panic, allowing the metrics server to maintain accurate reporting of all other metrics.

Reply:
gotcha